core/src/main/java/io/questdb/cairo/vm/MemoryPMARImpl.java

```diff
@@ -122,11 +124,8 @@ public void switchTo(int fd, long offset, byte truncateMode) {
     }
 
     public void sync(boolean async) {
-        if (pageAddress != 0) {
-            if (ff.msync(pageAddress, getExtendSegmentSize(), async) == 0) {
-                return;
-            }
-            LOG.error().$("could not msync [fd=").$(fd).$(", errno=").$(ff.errno()).$(']').$();
+        if (fd != -1) {
```

Comment:
We should also msync when a page gets released and a new one is being mmapped.

Reply:
done here, it is a little involved change because this memory uses sliding page. It maps/unmaps autonomously. Here: [MemoryPMARImpl.java](https://github.com/questdb/questdb/pull/3304/files/82e8eac6da0f25693d8bd701a86efd752ed80d1d#diff-46c653c0c19aad5d3db44df9a5cc68a2e78af47d1e0e3d1cf9b3d6636c569f8d)